### PR TITLE
Issue#45 Write an ADR about why we do open source development

### PR DESCRIPTION
**Pull request by RussellJimmies**
#45

### CODE REVIEW

**Review by k-allagbe — APPROVED** [review]
Concise.

Don't forget to rebase on the PR #41 and update the `index.md` file before merging.


Side consideration: add a descriptive tag to your branch names.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: adr/004-opensource-development.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Furthermore, our current technological landscape is heavily interwoven with open-source tools. The Government of Canada has also shown a commitment towards experimentation and open development, as reflected in platforms like alpha.canada.ca and the Digital Standards.
+
+
+### Solutions Considered
+**Full Open-Source Development**: Release all projects, without exception, to the public, encouraging community contribution and collaboration.
+
+**Hybrid Approach**: Develop core functionalities privately while making ancillary components or tools open source.
+
+**Private Development**: Keep all developments internal, only releasing tools and products after thorough vetting and quality assurance.
+
```
[the remark this answers is not in the document]
Comment (RussellJimmies) [reply adr/004-opensource-development.md:30]: Done

File: adr/004-opensource-development.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# ADR 4: Open-Source Development
+
+## Context
+The AI Lab is deliberating on the value proposition and risks of open-source development in government artificial intelligence projects. The goal is to ensure that we align our development efforts with the overall mission, values, and strategic objectives of the agency while also considering security, resource constraints, and quality.
```
[the remark this answers is not in the document]
Comment (rngadam) [reply adr/004-opensource-development.md:4]: @RussellJimmies you marked this as resolved but I don't see the fix

**Comment by rngadam** [comment]
@RussellJimmies I keep getting notified for review requested but I see some of the conversations haven't been addressed yet. Can you convert it to draft until you've addressed the required changes?

File: adr/004-opensource-development.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (RussellJimmies) [reply adr/004-opensource-development.md]: Done

File: adr/004-opensource-development.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+* Cost-effectiveness and avoiding vendor lock-ins.
+
+On the flip side, potential concerns include:
+
+* Exposing vulnerabilities, especially when sensitive information is involved.
+* Overhead of managing community contributions.
+* Potential quality control issues without the necessary expertise.
+* Political ramifications and criticism from stakeholders.
+
+Furthermore, our current technological landscape is heavily interwoven with open-source tools. The Government of Canada has also shown a commitment towards experimentation and open development, as reflected in platforms like alpha.canada.ca and the Digital Standards.
```
[the remark this answers is not in the document]
Comment (RussellJimmies) [reply adr/004-opensource-development.md]: Done

File: adr/004-opensource-development.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# ADR 4: Open-Source Development
+
+## Context
+The AI Lab is deliberating on the value proposition and risks of open-source development in government artificial intelligence projects. The goal is to ensure that we align our development efforts with the overall mission, values, and strategic objectives of the agency while also considering security, resource constraints, and quality.
+
+Open source development offers numerous advantages including:
```
[the remark this answers is not in the document]
Comment (RussellJimmies) [reply adr/004-opensource-development.md:6]: Done

File: adr/004-opensource-development.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
+# ADR 4: Open Source Development
+
+## Context
+The Canadian Food Inspection Food Agency (CFIA) Artificial Intelligence Lab's (AI Lab) mission is to push the boundaries of what AI is capable of doing. Our goal is to make it more accessible to CFIA employees by developing practical applications that help the CFIA fulfill its mandate. We are committed to open research and sharing our results with the broader AI community. In order to do so, opting for open source development is a must.
```
Comment (rngadam) [inline adr/004-opensource-development.md:4]: our mission is more in-line with "applied AI to the needs of the agency". we're an applied lab, not a research lab.

File: adr/004-opensource-development.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+2. **GNU General Public License (GPL)**: The GPL is designed to ensure that any adaptations or larger works based on a GPL-licensed project are also open source. It's often chosen to encourage the growth of open source ecosystems, though it can be restrictive for those wanting to incorporate the licensed code into proprietary software.
+
+3. **Apache License 2.0**: This is a permissive license similar to the MIT License but also provides an express grant of patent rights from contributors to users. It allows code use in proprietary software but requires preservation of the original copyright and license notices, along with a list of any significant changes made to the licensed material.
+
+4. **Creative Commons Attribution 4.0**: While not typically used for software, CC BY 4.0 is ideal for creative works. It allows sharing and adaptation of the material as long as appropriate credit is given, a link to the license is provided, and any changes made are indicated.
+
+
+## Decision
+After extensive evaluation, we have chosen to adopt the Full Open Source Development approach.
```
Comment (rngadam) [inline adr/004-opensource-development.md:50]: don't need "After extensive evaluation,"

File: adr/004-opensource-development.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+Every project undertaken by the AI Lab will be released publicly, embracing the ethos of open source development at its core. This means every aspect of our development – from core functionalities to ancillary tools – will be made available to the public, allowing for community contributions, feedback, and collaboration right from the inception of any project.
```
Comment (rngadam) [inline adr/004-opensource-development.md:52]: Every software product

**Review by rngadam — CHANGES_REQUESTED** [review]
format to linewrap at column 80.

Also write a blurb on how we handle sensitive data (public development vs private deployment)

File: adr/004-opensource-development.md (hunk 5 above)
Comment (RussellJimmies) [reply adr/004-opensource-development.md]: Reworded the context to reflect that

File: adr/004-opensource-development.md (hunk 6 above)
Comment (RussellJimmies) [reply adr/004-opensource-development.md]: Done

File: adr/004-opensource-development.md (hunk 7 above)
Comment (RussellJimmies) [reply adr/004-opensource-development.md]: Done

**Comment by RussellJimmies** [comment]
Converted to Draft, haven't formatted to linewrap at column 80

**Review by rngadam — CHANGES_REQUESTED** [review]
line-wrap your paragraph

File: adr/004-open-source-development.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+alpha.canada.ca and the Digital Standards.
+
+## Decision
+We have chosen to adopt the Full Open Source Development approach.
+
+Every software product undertaken by the AI Lab will be released publicly,
+embracing the ethos of open source development at its core. This means every
+aspect of our development – from core functionalities to ancillary tools – will
+be made available to the public, allowing for community contributions, feedback,
+and collaboration right from the inception of any project.
```
Comment (rngadam) [inline adr/004-open-source-development.md:55]: product

**Review by rngadam — CHANGES_REQUESTED** [review]

File: adr/004-open-source-development.md (hunk 8 above)
Comment (RussellJimmies) [reply adr/004-open-source-development.md]: Done

File: adr/004-open-source-development.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+insights and suggestions.
+
+**Potential Risks and Mitigation**: It's essential to acknowledge potential
+security risks associated with sensitive data. Proactive measures, including
+data anonymization, regular security audits, and stringent contribution
+guidelines, will be instituted to mitigate these risks.
+
+
+## References
+* [Government of Canada Digital Standards:
```
Comment (rngadam) [inline adr/004-open-source-development.md:144]: Couple more references to pages from well known organizations (Microsoft, OSI, FSF) on the "why Open Source"

**Review by rngadam — CHANGES_REQUESTED** [review]

File: adr/004-opensource-development.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rngadam) [reply adr/004-opensource-development.md]: add an heading explaining how we decided to standardize on open source into the ADR

File: adr/004-open-source-development.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+Open source development offers numerous advantages including:
+
+* Transparency and accountability to the citizens.
+* Encouraging collaboration and knowledge-sharing.
+* Tapping into collective intelligence for innovative solutions.
+* Peer-reviewed security enhancements.
+* Cost-effectiveness and avoiding vendor lock-ins.
+
+Moreover, Microsoft, who was opposed to the open source paradigm considering it a threat to its business, changed its stance significantly around 2010 when it began to embrace it. The company recognizes that "[building on the effort of others allows [them] to create meaningful value for [their] customers and engage with new ecosystems and user-bases in a natural way.](#ref2)"
```
Comment (rngadam) [inline adr/004-open-source-development.md:29]: this paragraph is not wrapped

File: adr/004-open-source-development.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+* Exposing vulnerabilities, especially when sensitive information is involved.
+* Overhead of managing community contributions.
+* Potential quality control issues without the necessary expertise.
+* Political ramifications and criticism from stakeholders.
+
+Additionally, our current technological landscape is heavily interwoven with open source tools. The Government of Canada has also shown a commitment towards
+experimentation and open development, as reflected in platforms like
+alpha.canada.ca and the [Digital Standards](#ref1).
+
+Finally, it's important to understand that "open source" doesn't solely refer to source code access, software must comply with the [Open Source Initiative's criteria](#ref3) to be **truly** open source. To put it simply, according to the Free Software Foundation, the users "[must have the freedom to run, edit, contribute to, and share the software.](#ref4)"
```
Comment (rngadam) [inline adr/004-open-source-development.md:48]: this paragraph is not wrapped

**Review by rngadam — CHANGES_REQUESTED** [review]

File: adr/004-open-source-development.md (hunk 10 above)
Comment (RussellJimmies) [reply adr/004-open-source-development.md:29]: Done

File: adr/004-open-source-development.md (hunk 11 above)
Comment (RussellJimmies) [reply adr/004-open-source-development.md:48]: Done

File: adr/004-open-source-development.md [hunk 12]
```diff
@@ -0,0 +1,169 @@
+# ADR-004: Open Source Development
+
+## Executive Summary
+The Canadian Food Inspection Food Agency (CFIA) Artificial Intelligence
+Laboratory (AI Lab) is adopting Full Open Source Development, releasing all
+software publicly to encourage transparent collaboration per the Government of
+Canada's standards. We've selected the MIT License for its flexibility,
+balancing open innovation with safeguarding sensitive data. This approach aims
+to harness collective expertise for enhanced innovation and trust in the digital
```
Comment (melanie-fressard) [inline adr/004-open-source-development.md:9]: weird format there

File: adr/004-open-source-development.md [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+* Transparency and accountability to the citizens.
+* Encouraging collaboration and knowledge-sharing.
+* Tapping into collective intelligence for innovative solutions.
+* Peer-reviewed security enhancements.
+* Cost-effectiveness and avoiding vendor lock-ins.
+
+Moreover, Microsoft, who was opposed to the open source paradigm considering it
+a threat to its business, changed its stance significantly around 2010 when it
+began to embrace it. The company recognizes that "[building on the effort of
+others allows [them] to create meaningful value for [their] customers and engage
```
Comment (melanie-fressard) [inline adr/004-open-source-development.md:32]: weird format here too

File: adr/004-open-source-development.md [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+* Overhead of managing community contributions.
+* Potential quality control issues without the necessary expertise.
+* Political ramifications and criticism from stakeholders.
+
+Additionally, our current technological landscape is heavily interwoven with
+open source tools. The Government of Canada has also shown a commitment towards
+experimentation and open development, as reflected in platforms like
+alpha.canada.ca and the [Digital Standards](#ref2).
+
+Finally, it's important to understand that "open source" doesn't solely refer to
```
Comment (melanie-fressard) [inline adr/004-open-source-development.md:53]: weird format hereee

File: adr/004-open-source-development.md [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+contribute to, and share the software.](#ref4)"
+
+
+## Decision
+We have chosen to adopt the Full Open Source Development approach.
+
+Every software product undertaken by the AI Lab will be released publicly,
+embracing the ethos of open source development at its core. This means every
+aspect of our development – from core functionalities to ancillary tools – will
+be made available to the public, allowing for community contributions, feedback,
```
Comment (melanie-fressard) [inline adr/004-open-source-development.md:66]: format

**Review by melanie-fressard — CHANGES_REQUESTED** [review]
There is a few format problems, you should re format it and it should be fine :) 
Good job, hype to know how to do the ref thingy

**Review by melanie-fressard — APPROVED** [review]

**Review by rngadam — APPROVED** [review]

**Review by leejaeka — APPROVED** [review]

**Review by rngadam — CHANGES_REQUESTED** [review]
rereading the internal 757/Why-OpenSource-development-at-the-AI-Lab, I see we're missing the whole section "specifics to our situation"  (our existing dependency on open source and publication on alpha, which should at leas tshow up as a reference).

**Review by redat97 — APPROVED** [review]
paradigm , learnt a new word !

**Review by CFIALeronB — APPROVED** [review]

**Review by JolanThomassin — APPROVED** [review]

**Comment by rngadam** [comment]
@RussellJimmies waiting on you addressing my last review comment

File: adr/004-open-source-development.md [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+* Political ramifications and criticism from stakeholders.
+
+Finally, it's important to understand that "open source" doesn't solely refer to
+source code access, software must comply with the [Open Source Initiative's
+criteria](#ref2) to be **truly** open source. To put it simply, according to the
+Free Software Foundation, the users "[must have the freedom to run, edit,
+contribute to, and share the software.](#ref3)"
+
+### Specifics to our situation
+#### Dependancy to Open Source
```
[the remark this answers is not in the document]
Comment (RussellJimmies) [reply adr/004-open-source-development.md:49]: That's now tattooed on my brain. 
Corrected.

**Review by rngadam — APPROVED** [review]

**Pull request merged**